File: app/src/main/java/com/lamnt/foodorder/view/adapter/OrdersAdapter.java
package com.lamnt.foodorder.view.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RatingBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.lamnt.foodorder.R;
import com.makeramen.roundedimageview.RoundedImageView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class OrdersAdapter extends RecyclerView.Adapter<OrdersAdapter.ViewHolder> {
    private LayoutInflater inflater;
    private Context context;
    private List<String> list;

    public OrdersAdapter(Context context, List<String> list) {
        this.context = context;
        inflater = LayoutInflater.from(context);
        this.list = list;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(inflater.inflate(R.layout.item_food_payment, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, final int position) {
        Glide.with(context).load(R.drawable.demo_tra_sua).placeholder(R.drawable.ic_food).into(holder.imgFood);
        String data = list.get(position);
        holder.txtAmount.setText(data);
        final int amount = Integer.parseInt(data);
        if (amount > 0) {
            holder.btnAdd.setOnClickListener(new View.OnClickListener() {
                @SuppressLint("SetTextI18n")
                @Override
                public void onClick(View v) {
                    holder.txtAmount.setText((amount + 1) + "");
                    list.set(position, (amount + 1) + "");
                    notifyItemChanged(position);
                }
            });

            holder.btnSub.setOnClickListener(new View.OnClickListener() {
                @SuppressLint("SetTextI18n")
                @Override
                public void onClick(View v) {
                    holder.txtAmount.setText((amount - 1) + "");
                    list.set(position, (amount - 1) + "");
                    notifyItemChanged(position);
                }
            });
        } else if (amount == 0) {
            list.remove(position);
            notifyDataSetChanged();
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.txt_restaurant_name)
        RoundedImageView txtRestaurantName;
        @BindView(R.id.cb_selected)
        CheckBox cbSelected;
        @BindView(R.id.img_food)
        RoundedImageView imgFood;
        @BindView(R.id.txt_price)
        TextView txtPrice;
        @BindView(R.id.ln_img)
        LinearLayout lnImg;
        @BindView(R.id.txt_name)
        TextView txtName;
        @BindView(R.id.rating_bar)
        RatingBar ratingBar;
        @BindView(R.id.txt_description)
        TextView txtDescription;
        @BindView(R.id.ln_text)
        LinearLayout lnText;
        @BindView(R.id.btn_add)
        ImageView btnAdd;
        @BindView(R.id.txt_amount)
        TextView txtAmount;
        @BindView(R.id.btn_sub)
        ImageView btnSub;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/lamnt/foodorder/view/adapter/recycleradapter/OrdersAdapter.kt
package com.lamnt.foodorder.view.adapter.recycleradapter

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.lamnt.foodorder.R
import kotlinx.android.synthetic.main.item_food_payment.view.*

class OrdersAdapter(
    private val context: Context,
    private var list: MutableList<String>
) : RecyclerView.Adapter<OrdersAdapter.ViewHolder>() {
    private val inflater = LayoutInflater.from(context)
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(inflater.inflate(R.layout.item_food_payment, parent, false))
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(position)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @SuppressLint("SetTextI18n")
        fun bind(position: Int) {
            itemView.apply {
                Glide.with(context)
                    .load(R.drawable.demo_tra_sua)
                    .placeholder(R.drawable.ic_food)
                    .into(imgFood!!)
                val data = list[position]
                txtAmount!!.text = data
                val amount = data.toInt()
                if (amount > 0) {
                    btnAdd!!.setOnClickListener {
                        txtAmount!!.text = (amount + 1).toString() + ""
                        list[position] = (amount + 1).toString() + ""
                        notifyItemChanged(position)
                    }
                    btnSub!!.setOnClickListener {
                        txtAmount!!.text = (amount - 1).toString() + ""
                        list[position] = (amount - 1).toString() + ""
                        notifyItemChanged(position)
                    }
                } else if (amount == 0) {
                    list.removeAt(position)
                    notifyDataSetChanged()
                }
            }
        }
    }

}